src/components/Filters/SchemaSieve.ts

```diff
@@ -1,157 +1,173 @@
-import {
-	AdvancedPineOperatorTest,
-	FilterModel,
-	FilterRule,
-	PineTypeModule,
-	Schema,
-	SchemaEntry,
-	SchemaSieveClass,
-} from 'rendition';
-import filterTests from '../PineTypes';
-
-type PineTypeModuleStruct = { [key: string]: PineTypeModule };
-
-export const SIMPLE_SEARCH_NAME = 'Full text search';
-
-class SchemaSieve implements SchemaSieveClass {
-	public SIMPLE_SEARCH_NAME = SIMPLE_SEARCH_NAME;
-	public tests: PineTypeModuleStruct;
-
-	constructor(tests: PineTypeModuleStruct = {}) {
-		this.tests = assign(cloneDeep(filterTests), tests);
-	}
-
-	test(item: any, input: FilterRule | FilterRule[]) {
-		return isArray(input)
-			? every(input, i => this.baseTest(item, i))
-			: this.baseTest(item, input);
-	}
+import { FilterSignature } from 'rendition';
+import * as utils from '../../utils';
+import DataTypes from '../DataTypes';
 
-	baseTest(item: any, input: FilterRule): boolean {
-		// If this is a compound rule, evaluate the rules together using OR logic
-		if (input.extra && input.extra.or) {
-			return (
-				this.baseTest(item, omit(input, 'extra') as FilterRule) ||
-				some(input.extra.or, rule => this.baseTest(item, rule as FilterRule))
-			);
-		}
+const ajv = new Ajv();
+ajvKeywords(ajv);
+ajv.addMetaSchema(metaSchema6);
 
-		const { type, name } = input;
-
-		// A simple search is not strictly a "type" and searches on all fields,
-		// so we handle that seperately here
-		if (name === this.SIMPLE_SEARCH_NAME) {
-			const { value } = input;
-			return some(
-				item,
-				target =>
-					target && includes(String(target).toLowerCase(), value.toLowerCase()),
-			);
-		}
+const FULL_TEXT_SLUG = 'full_text_search';
 
-		if (type in filterTests) {
-			const { operator, value } = input;
-			const target = item[name];
+export const filter = (
+	filters: JSONSchema6 | JSONSchema6[],
+	collection: any[],
+) => {
+	// Remove all schemas that may have been compiled already
+	ajv.removeSchema(/^.*$/);
 
-			if (operator! in filterTests[type].rules) {
-				const result = filterTests[type].rules[operator!];
-				return isPlainObject(result)
-					? result.test(target, value)
-					: result(target, value);
-			}
-		}
+	const validators = isArray(filters)
+		? filters.map(s => ajv.compile(s))
+		: [ajv.compile(filters)];
 
-		// If the rule contains an unrecognised type or operator, then play it safe
-		// and don't filter the item
-		return true;
+	if (isArray(collection)) {
+		return collection.filter(m => every(validators, v => v(m)));
 	}
 
-	filter<T>(
-		collection: T[],
-		input: FilterRule | FilterRule[],
-	): T[] | Partial<T> {
-		if (isArray(collection)) {
-			return this.filterArray(collection, input);
-		}
-		if (isPlainObject(collection)) {
-			return this.filterObject(collection, input) as Partial<T>;
-		}
+	return pickBy(collection, m => every(validators, v => v(m)));
+};
 
-		throw new Error('collection argument must be either object or array.');
+export const getDataModel = (schema?: JSONSchema6) => {
```

Comment:
> Would it make sense to move this to `DataTypes/index.tsx` so that it's closer to the objects that it manages/exposes? In that case the `DataTypes` folder/"namespace" would have the full responsibility of resolving and providing, eh, the DataTypes... and this file will be uncoupled from each specific DataType available.